incoming_test.go

```diff
@@ -74,6 +74,20 @@ func TestPeersIncomingConnection(t *testing.T) {
 				"incoming connections should not be added to non-root peer list")
 		}
 
+		// verify number of peers/connections on the client side
+		opts := &IntrospectionOptions{}
+		serverState := ch.IntrospectState(opts).RootPeers
+		for _, client := range []*Channel{ringpop, hyperbahn} {
+			clientPeerState := client.IntrospectState(opts).RootPeers
+			assert.Equal(t, len(clientPeerState), 1)
+			assert.Equal(t, len(clientPeerState[ch.PeerInfo().HostPort].OutboundConnections), 1)
+			assert.Equal(t, len(clientPeerState[ch.PeerInfo().HostPort].InboundConnections), 0)
+
+			assert.Equal(t, len(serverState), 2)
+			assert.Equal(t, len(serverState[client.PeerInfo().HostPort].InboundConnections), 1)
```

Comment:
nit: make a `clientHostPort` variable (it looks too similar to `ch.PeerInfo().HostPort` when it's so long). Do the same above, and maybe call that one `serverHostPort := ch.PeerInfo().HostPort`